nixos/modules/services/misc/paperless.nix

```diff
@@ -122,10 +122,12 @@ in
 
   config = mkIf cfg.enable {
 
-    systemd.tmpfiles.rules = [
-      "d '${cfg.dataDir}' - ${cfg.user} ${cfg.user} - -"
+    systemd.tmpfiles.rules = let
+      group = config.users.users.${cfg.user}.group;
+    in [
+      "d '${cfg.dataDir}' - ${cfg.user} ${group} - -"
     ] ++ (optional cfg.consumptionDirIsPublic
-      "d '${cfg.consumptionDir}' 777 ${cfg.user} ${cfg.user} - -"
+      "d '${cfg.consumptionDir}' 777 ${cfg.user} ${group} - -"
```

Comment:
If this directory is 777 anyway (which is an… interesting choice), why not just make it root:root so you don’t have to worry about users?

Reply:
Yeah, that's simpler and functionally equivalent.